code/modules/mob/living/simple_animal/eldritch_demons.dm

```diff
@@ -0,0 +1,281 @@
+	sight = SEE_MOBS|SEE_OBJS|SEE_TURFS
+	spells_to_add = list(/obj/effect/proc_holder/spell/targeted/ethereal_jaunt/shift/ash/long,/obj/effect/proc_holder/spell/targeted/telepathy/eldritch)
+
+/mob/living/simple_animal/hostile/eldritch/raw_prophet/Login()
+	. = ..()
+	client?.view_size.setTo(11)
+	//client?.change_view(11)
+
+/mob/living/simple_animal/hostile/eldritch/armsy
+	name = "Terror of the night"
+	real_name = "Armsy"
+	desc = "Abomination made from severed limbs."
+	icon_state = "armsy_start"
+	icon_living = "armsy_start"
+	maxHealth = 200
+	health = 200
+	melee_damage_lower = 10
+	melee_damage_upper = 15
+	///Previous segment in the chain
+	var/mob/living/simple_animal/hostile/eldritch/armsy/back
+	///Next segment in the chain
+	var/mob/living/simple_animal/hostile/eldritch/armsy/front
+	///Your old location
+	var/oldloc
+	///Allow / disallow pulling
+	var/allow_pulling = FALSE
+	///How many arms do we have to eat to expand?
+	var/stacks_to_grow = 5
+	///Currently eaten arms
+	var/current_stacks = 0
+
+/mob/living/simple_animal/hostile/eldritch/armsy/New(spawn_more = TRUE,len = 6)
+	. = ..()
+	len = max(3,len) //code breaks below 3, let's just not allow it.
+	oldloc = loc
+	if(!spawn_more)
+		return
+	allow_pulling = TRUE
+	var/mob/living/simple_animal/hostile/eldritch/armsy/next
+	var/mob/living/simple_animal/hostile/eldritch/armsy/prev
+	var/mob/living/simple_animal/hostile/eldritch/armsy/current
+	for(var/i in 0 to len)
+		prev = current
+		//i tried using switch, but byond is really fucky and it didnt work as intended. Im sorry
+		if(i == 0)
+			current = new type(drop_location(),spawn_more = FALSE)
+			current.icon_state = "armsy_mid"
+			current.icon_living = "armsy_mid"
+			current.front = src
+			current.AIStatus = AI_OFF
+			back = current
+		else if(i > 0 && i < len)
+			current = new type(drop_location(),spawn_more = FALSE)
+			prev.back = current
+			prev.icon_state = "armsy_mid"
+			prev.icon_living = "armsy_mid"
+			prev.front = next
+			prev.AIStatus = AI_OFF
+		else
+			prev.icon_state = "armsy_end"
+			prev.icon_living = "armsy_end"
+			prev.front = next
+			prev.AIStatus = AI_OFF
+		next = prev
+
+/mob/living/simple_animal/hostile/eldritch/armsy/can_be_pulled()
+	return FALSE
+
+/mob/living/simple_animal/hostile/eldritch/armsy/Moved()
+
+	if(back && back.loc != oldloc)
+		for(var/i in 1 to max(get_dist(back.loc,oldloc), 10) )
+			step_towards(back,oldloc)
+			if(back.loc == oldloc)
+				break
+		///ForceMove wont do, we must rebuild the worm, and moving there wont be enough, It MUST be teleported there or it WILL break. Why wont it work? because forceMove calls Move and it calls forceMove and so on and so on causing weird and janky behaviour. Please believe me i tried.
+		back.loc = oldloc
+
+	if(front && loc != front.oldloc)
+		for(var/i in 1 to max(get_dist(loc,front.oldloc), 10) )
+			step_towards(src,front.oldloc)
+			if(loc == front.oldloc)
+				break
+		loc = front.oldloc
+
+	oldloc = loc
+	gib_trail()
+	. = ..()
```

Comment:
Use `return ..()` at the end of the proc instead of `. = ..()`
Do the same on all instances.